### PR TITLE
Fix IDing and recurrence handling in GCalendar export

### PR DESCRIPTION
**Pull request by Ef55**
From #268 
```
Conversion (rightfully) fails on non-root recurring events.
Such events should be filtered out.
```

From #266 (go on dedicated issue for more details)
```
The generated IDs do not follow the guidelines [from GCalendar]. [...]
```

### CODE REVIEW

**Comment by Ef55** [comment]
Note: awful coverage doesn't come from those changes, but from the previous coverage of the changed class.

**Review by Sosol02 — APPROVED** [review]
Looks good, not much to say since this is a fix.

File: app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java [hunk 1]
```diff
@@ -53,18 +60,42 @@
...
     private GoogleCalendar() {
     }
 
+    /** Remove '-' from Id, as they are not allowed by GCalendar: https://developers.google.com/calendar/v3/reference/events/insert */
+    private static String fromId(Id id){
+        return id.getUuid().replace("-", "");
+    }
+
+    private static Id toId(String str) {
+        if (str.length() != Id.LENGTH - Id.SEP_COUNT) {
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java:77]: I assume this method will be called for converting GCalendar ids into Ids. Is it certain that the id format of all GCalendar ids verify this condition ?

Comment (Ef55) [reply app/src/main/java/com/github/onedirection/interoperability/gcalendar/GoogleCalendar.java:77]: Well, we are the one setting the IDs when exporting, so everything should be good.
Of course, if the user starts adding events into the calendar, then we don't control the IDs anymore, but that should be handled somewhere else.

**Review by Flechman — APPROVED** [review]
Thanks for rewritting the tests

**Pull request merged**